### PR TITLE
temp: log all layer loading errors while we lose them

### PR DESCRIPTION
**Pull request by koivunej**
Temporary workaround while some errors are not being logged.

Cc: #5815.

### CODE REVIEW

File: pageserver/src/tenant/storage_layer/layer.rs [hunk 1]
```diff
@@ -1211,8 +1211,10 @@ impl DownloadedLayer {
             // this will be a permanent failure
             .context("load layer");
 
-            if res.is_err() {
+            if let Err(e) = res.as_ref() {
                 LAYER_IMPL_METRICS.inc_permanent_loading_failures();
+                // TODO(#5815): we are not logging all errors, so temporarily log them here as well
+                tracing::error!("layer loading failed permanently: {e:#}");
```
Comment (jcsp) [inline pageserver/src/tenant/storage_layer/layer.rs:1218]: Maybe include the layer name in here in case the error doesn't?

**Review by jcsp — APPROVED** [review]

File: pageserver/src/tenant/storage_layer/layer.rs (hunk 1 above)
Comment (koivunej) [reply pageserver/src/tenant/storage_layer/layer.rs:1218]: With [f57b140](https://github.com/neondatabase/neon/pull/5816/commits/f57b1406ed590b337a9a04d702716c7fcb5daf8b)

**Comment by koivunej** [comment]
Will merge tomorrowish.

File: pageserver/src/tenant/storage_layer/layer.rs (hunk 1 above)
Comment (arpad-m) [reply pageserver/src/tenant/storage_layer/layer.rs:1218]: Regarding the newlines I was (wrongly) thinking about the debug impl. Thanks for explaining.

**Comment by koivunej** [comment]
> #### Failures on Postgres 16
> 
>     * `test_local_corruption`: [debug](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6787324315/index.html#suites/90de3f9cafdc78be9db0b2ada81f7c26/b260ae26dde3d973), [release](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6787324315/index.html#suites/90de3f9cafdc78be9db0b2ada81f7c26/b9c38f521ed5540d)
> 
> 
> #### Failures on Postgres 15
> 
>     * `test_local_corruption`: [release](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6787324315/index.html#suites/90de3f9cafdc78be9db0b2ada81f7c26/8cb59311ad2b24f2), [debug](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6787324315/index.html#suites/90de3f9cafdc78be9db0b2ada81f7c26/e33f091b9dd0fd85)
> 
> 
> #### Failures on Postgres 14
> 
>     * `test_local_corruption`: [debug](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6787324315/index.html#suites/90de3f9cafdc78be9db0b2ada81f7c26/32e80b107805d276), [release](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6787324315/index.html#suites/90de3f9cafdc78be9db0b2ada81f7c26/d1894cd9246da05f)

I really hate these stacktraces.

**Comment by koivunej** [comment]
Example from the stacktraceful log:

```
wal_connection_manager{tenant_id=114fd82e441050ccd0e42e54ca27dcff timeline_id=808198adfa4bdd1647f112254b2691cb}:connection{node_id=1}:get_value_reconstruct_data{layer=000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__00000000014EEA79-000000000152D319}: layer loading failed permanently: load layer: in-file summary does not match expected summary
```

**Comment by koivunej** [comment]
> #### Failures on Postgres 16
> 
> * `test_pageserver_lsn_wait_error_safekeeper_stop`: [debug](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6812819179/index.html#suites/420f58b116a84d3d8907f17743235f3d/a6c383191488aeb9)

Reported on #5154.

**Comment by koivunej** [comment]
> #### Failures on Postgres 16
> 
> * `test_branching_with_pgbench[cascade-1-10]`: [debug](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5816/6812819179/index.html#suites/ffbb7f9930a77115316b58ff32b7c719/65894e51247f0b7f)

This seems new.

**Pull request merged**